plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -173,101 +167,107 @@ export class PgInstrumentation extends InstrumentationBase {
+        // to properly narrow arg0, but TS 4.3.5 does not.
+        if (instrumentationConfig.addSqlCommenterCommentToQueries) {
+          args[0] = firstArgIsString
+            ? utils.addSqlCommenterComment(span, arg0 as string)
+            : firstArgIsQueryObjectWithText
+            ? {
+                ...(arg0 as ObjectWithText),
+                text: utils.addSqlCommenterComment(
+                  span,
+                  (arg0 as ObjectWithText).text
+                ),
+              }
+            : args[0];
         }
 
-        // Bind callback to parent span
+        // Bind callback (if any) to parent span (if any)
         if (args.length > 0) {
           const parentSpan = trace.getSpan(context.active());
           if (typeof args[args.length - 1] === 'function') {
             // Patch ParameterQuery callback
             args[args.length - 1] = utils.patchCallback(
-              plugin.getConfig(),
+              instrumentationConfig,
               span,
-              args[args.length - 1] as PostgresCallback
+              args[args.length - 1] as PostgresCallback // nb: not type safe.
             );
+
             // If a parent span exists, bind the callback
             if (parentSpan) {
               args[args.length - 1] = context.bind(
                 context.active(),
                 args[args.length - 1]
               );
             }
-          } else if (
-            typeof (args[0] as NormalizedQueryConfig).callback === 'function'
-          ) {
+          } else if (typeof queryConfig?.callback === 'function') {
             // Patch ConfigQuery callback
             let callback = utils.patchCallback(
               plugin.getConfig(),
               span,
-              (args[0] as NormalizedQueryConfig).callback!
+              queryConfig.callback as PostgresCallback // nb: not type safe.
             );
+
             // If a parent span existed, bind the callback
             if (parentSpan) {
               callback = context.bind(context.active(), callback);
             }
 
-            // Copy the callback instead of writing to args.callback so that we don't modify user's
-            // original callback reference
+            // Copy the callback instead of writing to args.callback so that we
+            // don't modify user's original callback reference
             args[0] = { ...(args[0] as object), callback };
           }
         }
 
-        // Perform the original query
-        const result: unknown = original.apply(this, args as any);
+        let result: unknown;
+        try {
+          result = original.apply(this, args as never);
+        } catch (e) {
+          // span.recordException(e);
+          span.setStatus({
+            code: SpanStatusCode.ERROR,
+            message:
+              typeof e === 'object' && e !== null
+                ? String((e as { message?: unknown }).message)
+                : undefined,
```

Comment:
> Can we replace with e?.message?

I don't think that's technically safe, because`e` could be a primitive value, and using `?.` with a primitive on the LHS would throw. Of course, it's quite unlikely that the thrown value would be a primitive, but there's nothing stopping that from being the case, so I figured we might as well be defensive about it.

Reply:
Ok, I made a utility function for this logic, to keep the main code path a little more readable, and so that I could reuse it in one other place where the code was assuming that the thrown value would be an `Error`.

Reply:
I think we tend to assume that thrown errors are not primitives in many other instrumentations, which simplifies the code, but I accepts your defensive approach :)